resources/views/standings/upload.blade.php

```diff
@@ -1,241 +1,3533 @@
+                
+                if(additionalDetailsStore.resultsPoints[i] !== 0) {
+                    resultsContent[i].points = additionalDetailsStore.resultsPoints[i];
+                }
+                
+                resultsContent[i].time = jsonResultsDetailsStore.raceTimeInAbsolutes[i];
+                
+                if(!isNaN(resultsContent[i].fastestlaptime)) {
+                    resultsContent[i].fastestlaptime = resultsContent[i].fastestlaptime.toFixed(3);
+                }
+            }
+        }
+    
+        return {track: trackContent[0], results: resultsContent};
+    }
+
+    function tableToJSON(table, supportingVariables) {
+        let data = [];
+        let headers = [];
+        let statusMap = [0, 1, -2, -3];
+
+        let jsonKeys = {
+            "Season": 'season_id',
+            "Round Number": 'round',
+            "Circuit": 'circuit_id',
+            "Points Scheme": 'points',
+            "Position": 'position',
+            "Driver": 'driver',
+            "Driver ID": 'driver_id',
+            "Constructor": 'constructor_id',
+            "Starting Grid": 'grid',
+            "Laps Completed": 'stops',
+            "Fastest Lap": 'fastestlaptime',
+            "Race Time": 'time',
+            "Status": 'status'
+        }
+
+        convertTableHeadersIntoRequiredKeyNames(headers, table, jsonKeys);
+
+        slotCellValuesIntoAppropriateHeaders(data, headers, table, supportingVariables);
+        
+        return data;
+    }
+
+    function convertTableHeadersIntoRequiredKeyNames(headers, table, jsonKeys) {
+        // Looping through cells of the first row of the table
+        for(let i = 0; i < table.rows[0].cells.length; i++) {
+            let tableHeaderText = table.rows[0].cells[i].innerText;
+
+            if(table.rows[0].cells[i].children[0] !== undefined) {
+                tableHeaderText = table.rows[0].cells[i].children[0].innerText;
+            }
+            
+            Object.keys(jsonKeys).forEach((key) => {
+                if(tableHeaderText == key) {
+                    headers[i] = jsonKeys[key];
+                }
+            });
+        }
+    }
+
+    function slotCellValuesIntoAppropriateHeaders(data, headers, table, supportingVariables) {
+        // Starting the loop from second row of the table to last row
+        for(let i = 1; i < table.rows.length; i++) {
+            let tableRow = table.rows[i];
+            let rowContent, rowContentFormat, treeTraversal, leftCellTraversal;
+            
+            let rowData = {};
+
+            // Looping through all cells of current row
+            for(let j = 0; j < tableRow.cells.length; j++) {
+                if(tableRow.cells[j].children[0].children.length != 0) {
+                    // Traversing through the DOM tree to find the cell value
+                    treeTraversal = tableRow.cells[j].children[0].children[0].options;
+
+                    if(headers[j] == 'driver'){
```

Comment:
use switch case insted of if else much more readable

Reply:
Note. Will transform code with switch case.